tests/test_metadata/test_v2.py

```diff
@@ -219,7 +217,7 @@ async def test_read_consolidated_metadata(
                     fill_value=0,
                     chunks=(730,),
                     attributes={"_ARRAY_DIMENSIONS": ["time"], "dataset": "NMC Reanalysis"},
-                    dtype=np.dtype("int16"),
+                    dtype=Int16(),
```

Comment:
This is a breaking API change right, because the type of `dtype` that is returned in the `ArrayV2Metadata` object has changed? If so, this needs a clear changelog entry communicating this change.

Reply:
yes it is breaking, and to be expected since the goal if this PR is to introduce a new model for data types which necessarily replaces the old one. I can make this more clear in the release notes.